### DISCUSSION TITLE
Implement Rimworld Stuffables

### DISCUSSION THREAD

**Discussion by ghost** [cataclysmbn/Cataclysm-BN#3504] [General]
If you are unfamiliar with Rimworld this is the core concept:
https://rimworldwiki.com/wiki/Stuff

How we might go about implementing stuffables:
1. A clothing item will have a flag for what type of material it is made of, cloth, leather, etc. We can refine this.
2. The flag from 1 will determine allowed items to craft it. Cloth material can allow nylon, rags, kevlar, etc. The volume of the clothing must be met in the recipe. The recipe will dynamically change per material so if rags are 100mL and kevlar is 50mL it would take twice as much kevlar.
3. A clothing item will be flagged with its material and inherit its weight and material and thus protection from what was used to craft it. 
4. We then can remove every clothing variation such as wool t shirts, kevlar t shirts, and instead spawn the default item made from the default material, with an optional way to define material in the itemgroup definition. This will tremendously cut back on item count.

For weapons: It is a bit trickier unless we add a material definition for less/more damage done if made of a material. Lead might be better for blunt weapons, and a material could be assumed as default in a weapon so steel will be the case in most examples.

**Comment by ghost** [comment]
Edit: I forgot about warmth. A material could have a default defined warmth value in its material config,  material thickness on a clothing item would multiply this the same way as how protection is calculated based on material protection and thickness.

**Comment by Zlorthishen** [comment]
given how `materials` are defined in game, this seems interesting since alot of the groundwork is already laid

**Comment by Unrepentant-Atheist** [comment]
The core concept of Rimworld is...tables and hats made out of human skins.
